### ISSUE TITLE
Soft object type is incompatible

### ISSUE THREAD

**Issue by sergeysova** [JSMonk/hegel#67]
```ts
const softUser: { email: string, password: string, ... } = { // Error here
  name: "any.user@gmail.com",
  password: "qwerty",
  status: "offline"
};
```

```
Type "{
	name: 'any.user@gmail.com',
	password: 'qwerty',
	status: 'offline'
}" is incompatible with type "{ email: string, password: string, ... }"
```

[Playground](https://jsmonk.github.io/hegel/try#MYewdgzgLgBBIDMoFUIFMBOAuGBvGaAtgIYCWANjtBqWAOYA0MADsRBAO4gYAmVUNekwB0omAF8YAXjwwA9HJiBeDcAyOwEIAUDBhhihNDgBExMAE9hAV3QYAAnRIVhoQoYZaWbTtz4xDARw5MKFNXd2hiKCsjRARyWjRDDXEAbg0gA)

Example from [Strict and Soft object types](https://jsmonk.github.io/hegel/docs/object-types#strict-and-soft-object-types)

**Comment by JSMonk** [comment]
Were fixed at 8160a4c2b0acc4c070bfa836943427fd1780f153

**Comment by JSMonk** [comment]
Thank you for contribution ^_^
